hi-ml/src/health_ml/utils/reports.py

```diff
@@ -149,47 +150,67 @@ def add_text(self, text: str, tag_class: str = '') -> None:
+        a list of path to one or more .csv files, or a directory of csv files containing your tables.
+        If neither of these parameters are provided, an Exception will be raised.
 
         :param table: An optional Pandas DataFrame to be rendered in the report
-        :param table_path: An optional path to a .csv file containing the table to be rendered on the report
+        :param table_paths_or_dir: An optional path to a .csv file containing the table to be rendered on the report
         :raises ValueError: If neither a table object nor a path to a .csv file are provided
         """
-        if table is None and table_path is None:
+        if table is None and len(table_paths_or_dir) == 0:
             raise ValueError("One of table or table path must be provided")
-        if table is None:
-            table = pd.read_csv(table_path)
 
-        num_existing_tables = self.template.count("table.to_html")
-        table_key = f"{TABLE_KEY_HTML}_{num_existing_tables}"  # starts at zero
+        tables = []
+        if table is not None:
+            tables.append(table)
 
-        template_addition = """<div class="container" >
-        {% for table in """ + table_key + """ %}
-            {{ table.to_html(classes=[ "table"], justify="center") | safe }}
-        {% endfor %}
-        </div>
-        <br>"""
-        self.add_to_template(template_addition)
+        if len(table_paths_or_dir) > 0:
+            for table_path_or_dir in table_paths_or_dir:
+                table_path_or_dir = Path(table_path_or_dir)
+                if table_path_or_dir.is_dir():
+                    for table_path in table_path_or_dir.iterdir():
+                        self.add_tables(table_paths_or_dir=[table_path])
+                else:
+                    tables.append(pd.read_csv(table_path_or_dir))
 
-        self.render_kwargs.update({table_key: [table]})
+        self._add_tables_to_report(tables)
 
-    def add_image(self, image_path: str, figsize: Tuple[int, int] = DEFAULT_FIGSIZE) -> None:
+    def _add_image_to_report(self, image_path: Path, base64_encode: bool = False) -> None:
         """
-        Given a path to an image file, embeds the image on the report. If the path is within the report folder, the
-        relative path will be used. This is to ensure that the HTML document is able to locate and embed the image.
-        Otherwise, the image path is not altered.
+        Given a path to an image, add it to the report template and to the report arguments
+        for rendering later. Optionally encode as base64 - this is useful if a standalone
+        report is required but leads to a much larger report file size, so is False by default
 
-        :param image_path: The path to the image to be embedded
-        :param figsize: The size of the figure
+        :param image_path: The paths to the image to be added
+        :param base64_encode: If True, encode the image as base64 in the HTML report. Default is False
```

Comment:
this sounds as you have tried to encode the tiles images but this was leading to a too large file - is it true?

Reply:
It does lead to pretty large files -the histopathology 1st gen report with encoded images is about 84MB without any compression with encoded images